### PR TITLE
Step 3 Pages Routing

### PR DESCRIPTION
**Pull request by a-reznikov**
routing
details

- [x] Create new page that will be used to view full information about story
- [x] Create link from main page from stories to story page
- [x] Create commit and PR to main branch
- [x] Send link to PR to your lead in slack

### CODE REVIEW

File: src/api/news/queries.ts [hunk 1]
```diff
@@ -5,5 +5,7 @@ export const useNews = (sectionName: string) => {
   return useQuery({
     queryKey: ["section", sectionName],
     queryFn: () => getNewsBySection(sectionName),
+    enabled: sectionName !== "[section]",
+    refetchOnWindowFocus: false,
```
Comment (Ilya-Valasiuk) [inline src/api/news/queries.ts]: check

File: src/api/news/queries.ts [hunk 2]
```diff
@@ -5,5 +5,7 @@ export const useNews = (sectionName: string) => {
   return useQuery({
     queryKey: ["section", sectionName],
```
Comment (Ilya-Valasiuk) [inline src/api/news/queries.ts]: enum QueryKeys {
  SECTION = 'section',
}

File: src/components/common/header/Header.tsx [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
 import classNames from "classnames";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
 import { HEADER_NAVIGATION } from "@/constants/links";
 import Logo from "/public/icons/logo.svg";
 import Menu from "/public/icons/menu.svg";
 
 export const Header: React.FC = () => {
+  const router = useRouter();
+  const [, sectionFromUrl] = router.asPath.split("/");
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: check

File: src/styles/globals.css [hunk 4]
```diff
@@ -1,3 +1,11 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
+.links {
```
Comment (Ilya-Valasiuk) [inline src/styles/globals.css]: move to custom Link component

File: src/pages/[section]/index.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { Home } from "@/components/home/Home";
+
+export default function SectionPage() {
```
Comment (Ilya-Valasiuk) [inline src/pages/[section]/index.tsx]: slug

File: src/components/common/header/Header.tsx (hunk 3 above)
Comment (Ilya-Valasiuk) [reply src/components/common/header/Header.tsx]: router.query.section

File: src/api/news/queries.ts (hunk 2 above)
Comment (a-reznikov) [reply src/api/news/queries.ts]: Created

File: src/api/news/queries.ts (hunk 1 above)
Comment (a-reznikov) [reply src/api/news/queries.ts]: Implemented for QueryClient

File: src/components/common/header/Header.tsx (hunk 3 above)
Comment (a-reznikov) [reply src/components/common/header/Header.tsx]: Used

File: src/styles/globals.css (hunk 4 above)
Comment (a-reznikov) [reply src/styles/globals.css]: Implemented

File: src/pages/[section]/index.tsx (hunk 5 above)
Comment (a-reznikov) [reply src/pages/[section]/index.tsx]: Used

File: src/api/articles/formatter.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { formateDate } from "@/utils/formatter-date";
+import { ArticleRaw, Article, ArticleImages } from "./types";
+import { API_BASE_IMG_URL } from "@/constants/variables";
+
+export const prepareArticle = (article: ArticleRaw[]): Article | null => {
+  if (!article.length) return null;
```
Comment (Ilya-Valasiuk) [inline src/api/articles/formatter.ts]:
```
if (!article.length) {
  return null;
}
```

File: src/components/common/custom-link/CustomLink.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import Link from "next/link";
+import classNames from "classnames";
+
+type LinksProps = {
```
Comment (Ilya-Valasiuk) [inline src/components/common/custom-link/CustomLink.tsx]:
```
type Props
```

File: src/components/common/custom-link/CustomLink.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  href: string;
+  className: string;
+};
+
+export const CustomLink: React.FC<React.PropsWithChildren & LinksProps> = ({
```
Comment (Ilya-Valasiuk) [inline src/components/common/custom-link/CustomLink.tsx]: where other props from Link?

File: src/components/common/footer/Footer.tsx [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import classNames from "classnames";
 import { FOOTER_NAVIGATION, SOCIAL_LINKS } from "@/constants/links";
 import React from "react";
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: check lint

File: src/components/common/footer/Footer.tsx [hunk 10]
```diff
@@ -23,23 +24,23 @@ export const Footer: React.FC = () => {
           )}
         >
           {FOOTER_NAVIGATION.map(({ name, href }) => (
-            <a key={name} href={href}>
-              {name}
-            </a>
+            <div key={name}>
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: ?

File: src/components/common/custom-link/CustomLink.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  children,
+  href,
+  className,
```
Comment (Ilya-Valasiuk) [inline src/components/common/custom-link/CustomLink.tsx:13]: ...otherProps

File: src/components/common/custom-link/CustomLink.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+}) => {
+  return (
+    <Link
+      href={href}
+      className={classNames(
+        "transition duration-300 ease-out hover:ease-in hover:text-main",
+        className
+      )}
```
Comment (Ilya-Valasiuk) [inline src/components/common/custom-link/CustomLink.tsx:22]: {...otherProps}

File: src/components/common/footer/Footer.tsx [hunk 13]
```diff
@@ -23,23 +24,23 @@ export const Footer: React.FC = () => {
...
-              key={alt}
-              href={href}
-              className={classNames(
-                "w-10 h-10 transition duration-300 ease-out hover:ease-in hover:text-main",
-                "md:w-6 md:h-6"
-              )}
-            >
-              {React.createElement(component)}
-            </a>
+            <div key={alt} className="flex">
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: ?

File: src/components/common/header/Header.tsx [hunk 14]
```diff
@@ -1,9 +1,25 @@
...
+import { useEffect, useState } from "react";
 import { HEADER_NAVIGATION } from "@/constants/links";
+import { hasSlugStringType } from "@/utils/type-guards";
 import Logo from "/public/icons/logo.svg";
 import Menu from "/public/icons/menu.svg";
+import { CustomLink } from "../custom-link/CustomLink";
 
 export const Header: React.FC = () => {
+  const router = useRouter();
+  const sectionName: string | string[] | undefined = router.query.section;
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: const sectionName = router.query.section;

File: src/components/common/header/Header.tsx [hunk 15]
```diff
@@ -1,9 +1,25 @@
...
+  const [currentSection, setCurrentSection] = useState("");
+
+  useEffect(() => {
+    if (router.isReady && hasSlugStringType(sectionName)) {
+      setCurrentSection(sectionName);
+    } else if (router.isReady && !hasSlugStringType(sectionName)) {
+      setCurrentSection("home");
+    }
+  }, [router.isReady, sectionName]);
+
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: ?

File: src/components/common/header/Header.tsx [hunk 16]
```diff
@@ -12,40 +28,34 @@ export const Header: React.FC = () => {
...
-                {
-                  "text-dark border-b-main": current,
-                }
-              )}
-              aria-current={current ? "page" : undefined}
-            >
-              {name}
-            </a>
+          {HEADER_NAVIGATION.map(({ name, href }) => (
+            <div key={name} className="h-full flex">
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: ?

File: src/types/index.ts [hunk 17]
```diff
@@ -0,0 +1,3 @@
+export type PropsWithSlug = {
+  section: string | string[] | undefined;
+};
```
Comment (Ilya-Valasiuk) [inline src/types/index.ts]: not needed

File: src/components/common/news-list/NewsList.tsx [hunk 18]
```diff
@@ -1,15 +1,18 @@
...
+import { hasSlugStringType } from "@/utils/type-guards";
+import { PropsWithSlug } from "@/types";
 
-type Props = {
-  section: string;
-};
-
-export const NewsList: React.FC<Props> = ({ section }) => {
-  const { isPending, isError, data, error } = useNews(section);
+export const NewsList: React.FC<PropsWithSlug> = ({ section }) => {
```
Comment (Ilya-Valasiuk) [inline src/components/common/news-list/NewsList.tsx]: type Propss

File: src/components/common/header/Header.tsx (hunk 16 above)
Comment (a-reznikov) [reply src/components/common/header/Header.tsx]: Fixed

File: src/components/common/footer/Footer.tsx (hunk 13 above)
Comment (a-reznikov) [reply src/components/common/footer/Footer.tsx]: Fixed

File: src/api/articles/formatter.ts (hunk 6 above)
Comment (a-reznikov) [reply src/api/articles/formatter.ts]: Fixed

File: src/components/common/custom-link/CustomLink.tsx (hunk 7 above)
Comment (a-reznikov) [reply src/components/common/custom-link/CustomLink.tsx]: Changed

File: src/components/common/custom-link/CustomLink.tsx (hunk 8 above)
Comment (a-reznikov) [reply src/components/common/custom-link/CustomLink.tsx]: Added

File: src/components/common/footer/Footer.tsx (hunk 9 above)
Comment (a-reznikov) [reply src/components/common/footer/Footer.tsx]: Checked

File: src/components/common/footer/Footer.tsx (hunk 10 above)
Comment (a-reznikov) [reply src/components/common/footer/Footer.tsx]: Fixed

File: src/components/common/custom-link/CustomLink.tsx (hunk 11 above)
Comment (a-reznikov) [reply src/components/common/custom-link/CustomLink.tsx:13]: Used

File: src/components/common/custom-link/CustomLink.tsx (hunk 12 above)
Comment (a-reznikov) [reply src/components/common/custom-link/CustomLink.tsx:22]: Used

File: src/components/common/header/Header.tsx (hunk 14 above)
Comment (a-reznikov) [reply src/components/common/header/Header.tsx]: Fixed

File: src/components/common/header/Header.tsx (hunk 15 above)
Comment (a-reznikov) [reply src/components/common/header/Header.tsx]: Discussed

File: src/types/index.ts (hunk 17 above)
Comment (a-reznikov) [reply src/types/index.ts]: Deleted

File: src/components/common/news-list/NewsList.tsx (hunk 18 above)
Comment (a-reznikov) [reply src/components/common/news-list/NewsList.tsx]: Changed

File: src/components/common/custom-link/CustomLink.tsx [hunk 19]
```diff
@@ -0,0 +1,28 @@
+import Link, { LinkProps } from "next/link";
+import classNames from "classnames";
+
+type Props = React.PropsWithChildren &
+  LinkProps & {
+    href: string;
+    className: string;
+  };
+
+export const CustomLink: React.FC<Props> = ({
```
Comment (Ilya-Valasiuk) [inline src/components/common/custom-link/CustomLink.tsx:10]: export const CustomLink: React.FC<React.PropsWithChildren<LinkProps>> = ({

**Pull request merged**